buildpacktoml_test.go

```diff
@@ -0,0 +1,745 @@
+						BOM: nil,
+						Labels: []lifecycle.Label{
+							{Key: "some-key", Value: "some-value"},
+							{Key: "some-other-key", Value: "some-other-value"},
+						},
+						MetRequires: nil,
+						Processes:   []launch.Process{},
+						Slices:      []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+
+				it("should include processes", func() {
+					mkfile(t,
+						`[[processes]]`+"\n"+
+							`type = "some-type"`+"\n"+
+							`command = "some-cmd"`+"\n"+
+							`[[processes]]`+"\n"+
+							`type = "other-type"`+"\n"+
+							`command = "other-cmd"`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM:         nil,
+						Labels:      []lifecycle.Label{},
+						MetRequires: nil,
+						Processes: []launch.Process{
+							{Type: "some-type", Command: "some-cmd", BuildpackID: "A"},
+							{Type: "other-type", Command: "other-cmd", BuildpackID: "A"},
+						},
+						Slices: []layers.Slice{},
+					}); s != "" {
+						t.Fatalf("Unexpected metadata:\n%s\n", s)
+					}
+				})
+
+				it("should include slices", func() {
+					mkfile(t,
+						"[[slices]]\n"+
+							`paths = ["some-path", "some-other-path"]`+"\n",
+						filepath.Join(appDir, "launch-A-v1.toml"),
+					)
+
+					br, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+					if err != nil {
+						t.Fatalf("Unexpected error:\n%s\n", err)
+					}
+
+					if s := cmp.Diff(br, lifecycle.BuildResult{
+						BOM:         nil,
+						Labels:      []lifecycle.Label{},
+						MetRequires: nil,
+						Processes:   []launch.Process{},
+						Slices:      []layers.Slice{{Paths: []string{"some-path", "some-other-path"}}},
+					}); s != "" {
+						t.Fatalf("Unexpected:\n%s\n", s)
+					}
+				})
+			})
+		})
+
+		when("building succeeds with a clear env", func() {
+			it.Before(func() {
+				env.EXPECT().List().Return(append(os.Environ(), "TEST_ENV=cleared"))
+
+				bpTOML.Buildpack.Version = "v1.clear"
+				bpTOML.Path = filepath.Join(buildpacksDir, "A", "v1.clear")
+				bpTOML.Buildpack.ClearEnv = true // TODO: it is weird that all three of these conditions need to be specified
+			})
+
+			it("should not apply user-provided env vars", func() {
+				if _, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config); err != nil {
+					t.Fatalf("Error: %s\n", err)
+				}
+				if s := cmp.Diff(rdfile(t, filepath.Join(appDir, "build-info-A-v1.clear")),
+					"TEST_ENV: cleared\n",
+				); s != "" {
+					t.Fatalf("Unexpected info:\n%s\n", s)
+				}
+			})
+
+			it("should set CNB_BUILDPACK_DIR", func() {
+				if _, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config); err != nil {
+					t.Fatalf("Error: %s\n", err)
+				}
+				bpsDir, err := filepath.Abs(buildpacksDir) // TODO: this should be passed in somehow
```

Comment:
This feels kind of odd, it would be nice to derive this from `bpTOML.Path` instead of `buildpacksDir` which is even higher in the test setup. But that seemed to introduce more complexity than I'd like to deal with at the moment...